packages/valory/skills/transaction_settlement_abci/payloads.py

```diff
@@ -41,226 +42,66 @@ def __str__(self) -> str:
-        """Get the randomness."""
-        return self._randomness  # pragma: nocover
-
-    @property
-    def data(self) -> Dict:
-        """Get the data."""
-        return dict(round_id=self.round_id, randomness=self.randomness)
-
 
+@dataclass(frozen=True)
 class SelectKeeperPayload(BaseTxPayload):
     """Represent a transaction payload of type 'select_keeper'."""
 
+    keepers: str
     transaction_type = TransactionType.SELECT_KEEPER
 
-    def __init__(self, sender: str, keepers: str, **kwargs: Any) -> None:
-        """Initialize a 'select_keeper' transaction payload.
-
-        :param sender: the sender (Ethereum) address
-        :param keepers: the updated keepers
-        :param kwargs: the keyword arguments
-        """
-        super().__init__(sender, **kwargs)
-        self._keepers = keepers
-
-    @property
-    def keepers(self) -> str:
-        """Get the keeper."""
-        return self._keepers
-
-    @property
-    def data(self) -> Dict[str, str]:
-        """Get the data."""
-        return dict(keepers=self.keepers)
-
 
+@dataclass(frozen=True)
 class ValidatePayload(BaseTxPayload):
     """Represent a transaction payload of type 'validate'."""
 
+    vote: Optional[bool] = None
     transaction_type = TransactionType.VALIDATE
 
-    def __init__(self, sender: str, vote: Optional[bool] = None, **kwargs: Any) -> None:
-        """Initialize an 'validate' transaction payload.
-
-        :param sender: the sender (Ethereum) address
-        :param vote: the vote
-        :param kwargs: the keyword arguments
-        """
-        super().__init__(sender, **kwargs)
-        self._vote = vote
-
-    @property
-    def vote(self) -> Optional[bool]:
-        """Get the vote."""
-        return self._vote
-
-    @property
-    def data(self) -> Dict:
-        """Get the data."""
-        return dict(vote=self.vote) if self.vote is not None else {}
-
 
+@dataclass(frozen=True)
 class CheckTransactionHistoryPayload(BaseTxPayload):
     """Represent a transaction payload of type 'check'."""
 
+    verified_res: str
     transaction_type = TransactionType.CHECK
 
-    def __init__(self, sender: str, verified_res: str, **kwargs: Any) -> None:
-        """Initialize an 'check' transaction payload.
-
-        :param sender: the sender (Ethereum) address
-        :param verified_res: the verification result
-        :param kwargs: the keyword arguments
-        """
-        super().__init__(sender, **kwargs)
-        self._verified_res = verified_res
-
-    @property
-    def verified_res(self) -> str:
-        """Get the verified result."""
-        return self._verified_res
-
-    @property
-    def data(self) -> Dict:
-        """Get the data."""
-        return dict(verified_res=self.verified_res)
-
 
+@dataclass(frozen=True)
 class SynchronizeLateMessagesPayload(BaseTxPayload):
     """Represent a transaction payload of type 'synchronize'."""
 
+    tx_hashes: tuple[str]
     transaction_type = TransactionType.SYNCHRONIZE
```

Comment:
this transaction_type will be deleted as well later on. First addressing the rounds now, then the behaviours